### PR TITLE
Implement/exception 2

### PR DESCRIPTION
**Pull request by rdrdrdrd95**
Reworked from more recent master with a less invasive implmentation of exception popups. 

Renamed all exceptions to e vs e, err, and ex

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
You shouldn't be touching the third party folder as part of this.

File: src/OpenLoco/Input/Keyboard.cpp [hunk 1]
```diff
@@ -500,6 +500,7 @@ namespace OpenLoco::Input
                 catch (const std::exception&)
                 {
                     Windows::showError(StringIds::screenshot_failed);
+                    Ui::showMessageBox("Exception:", "screenshot_failed");
```
Comment (duncanspumpkin) [inline src/OpenLoco/Input/Keyboard.cpp:503]: If an error box is being shown we don't need an additional system error box. This is for situations where we can't use the normal error box

**Comment by duncanspumpkin** [comment]
I'm not sure all of these make sense. We only want to resort to the system error box when we can't use the normal ui. So that will be any situation that is forcing the game to stop or a situation where the ui hasn't loaded yet.

**Comment by duncanspumpkin** [comment]
Also in general just work from the one PR. I take it i can close the old one now?

**Comment by rdrdrdrd95** [comment]
Reverted the 3rd party files, accidentally ran formatting on those. Also removed the one popup that was redundant.

File: src/OpenLoco/OpenLoco.cpp [hunk 2]
```diff
@@ -951,6 +952,7 @@ namespace OpenLoco
         catch (const std::exception& e)
         {
             std::fprintf(stderr, "Unable to autosave game: %s\n", e.what());
+            Ui::showMessageBox("Unable to Autosave:", e.what());
```
Comment (duncanspumpkin) [inline src/OpenLoco/OpenLoco.cpp:955]: Is this actually worthy of a showMessasgeBox. Autosaves only happen when you are already in game and everything is loaded. Therefore the normal interface can be used to show the error.

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp [hunk 3]
```diff
@@ -763,9 +763,10 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
                         _newFiles.emplace_back(name, isDirectory);
                     }
                 }
-                catch (const fs::filesystem_error& err)
+                catch (const fs::filesystem_error& e)
                 {
-                    Console::error("Invalid directory or file: %s", err.what());
+                    Console::error("Invalid directory or file: %s", e.what());
+                    Ui::showMessageBox("Exception: ", e.what());
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/PromptBrowseWindow.cpp:769]: Again don't think this needs an showMessageBox this is a window so clearly the normal window ui can be used.

**Comment by duncanspumpkin** [comment]
I still think there are some exceptions that dont need this and can use the normal error message mechanism. Screenshot failure for example would only ever make sense after the ui has loaded so therefore using the normal error message makessense. S5 loading, localisation loading, and the main loop exception all make sense those are key things that can happen prior to loading the Ui. The others perhaps not required.

**Comment by rdrdrdrd95** [comment]
I wnet through this again and tried to find situations where these errors would occur and pruned some of the ones that may make sense to use the normal UI. 

For the time being I will leave these as is, keeping just the 'more fatal' errors as popups. 

The one error I am keeping that I think is 'more' fatal is failure to save, since you will absolutely want to know this when you try to save in as many ways as possible.

File: src/OpenLoco/OpenLoco.cpp (hunk 2 above)
Comment (rdrdrdrd95) [reply src/OpenLoco/OpenLoco.cpp]: removed this

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp (hunk 3 above)
Comment (rdrdrdrd95) [reply src/OpenLoco/Windows/PromptBrowseWindow.cpp]: removed this

File: src/OpenLoco/S5/S5.cpp [hunk 4]
```diff
@@ -371,6 +371,7 @@ namespace OpenLoco::S5
         catch (const std::exception& e)
         {
             std::fprintf(stderr, "Unable to save S5: %s\n", e.what());
+            Ui::showMessageBox("Unable to save S5", e.what());
```
Comment (AaronVanGeffen) [inline src/OpenLoco/S5/S5.cpp:374]: I think we can use an in-game error window for any S5 related loading errors. @duncanspumpkin, any thoughts?

**Review by AaronVanGeffen — APPROVED** [review]

File: src/OpenLoco/S5/S5.cpp (hunk 4 above)
Comment (duncanspumpkin) [reply src/OpenLoco/S5/S5.cpp]: Hmm not too sure. What would happen if title screen failed to load? I guess that doesn't mean the ui has failed to load so could present an error but not too sure.

Comment (AaronVanGeffen) [reply src/OpenLoco/S5/S5.cpp]: If the title screen fails to load, you'll still get a game window:

Screenshot (3)

**Comment by duncanspumpkin** [comment]
Appears to work. Didn't try it with breakpad but assume that works with it.

**Pull request merged**